test/resignation.test.ts

```diff
@@ -136,18 +123,57 @@ describe("LiquidityBridgeContractV2 resignation process should", () => {
 
     await expect(withdrawCollateralTx)
       .to.emit(lbc, "WithdrawCollateral")
-      .withArgs(provider.signer.address, LP_COLLATERAL);
+      .withArgs(liquidityProvider.signer.address, LP_COLLATERAL);
     await lpBalanceAfterCollateralWithdrawAssertion(
       LP_COLLATERAL - withdrawCollateralReceipt!.fee
     );
     await peginCollateralAfterWithdrawAssertion();
     await pegoutCollateralAfterWithdrawAssertion();
     await lbcBalanceAfterCollateralWithdrawAssertion();
     await expect(
-      lbc.getCollateral(provider.signer.address)
+      lbc.getCollateral(liquidityProvider.signer.address)
     ).to.eventually.be.eq(0);
     await expect(
-      lbc.getPegoutCollateral(provider.signer.address)
+      lbc.getPegoutCollateral(liquidityProvider.signer.address)
     ).to.eventually.be.eq(0);
   });
+
+  it("resign as pegout only, pegin only and both", async () => {
```

Comment:
I think you should include in the test that they all can withdraw the collateral, because that was what the report was complaining about

Reply:
you could also split the test in 3 and do that in the 3 tests, resignation + collateral withdrawal